lib/transport.js

```diff
@@ -14,7 +14,20 @@ class Transport {
 
   connect(cb) {
     return this._openSocket().then(() => this._authenticate()).then(() => {
-      this.socket.on('data', cb)
+      var buffers = []
+      var rem, len
+      this.socket.on('data', (chunk) => {
```

Comment:
This works great but I was assuming that transport.js would be protocol agnostic so instead I would do exact same thing in client.js. But then again if you are happy with this I do not mind :)